src/mem/AllocatorProxy.h

```diff
@@ -48,43 +46,29 @@ namespace Mem
 /**
  * Support late binding of pool type for allocator agnostic classes
  */
-class AllocatorProxy
+class AllocatorProxy : public Allocator
 {
 public:
     AllocatorProxy(char const *aLabel, size_t const &aSize, bool doZeroBlocks = true):
-        label(aLabel),
+        Allocator(aLabel, doZeroBlocks),
         size(aSize),
-        theAllocator(nullptr),
-        doZero(doZeroBlocks)
+        theAllocator(nullptr)
     {}
 
-    /// Allocate one element from the pool
-    void *alloc();
-
-    /// Free a element allocated by Mem::AllocatorProxy::alloc()
-    void freeOne(void *);
-
-    int inUseCount() const;
-    size_t objectSize() const {return size;}
-    char const * objectType() const {return label;}
-
-    MemPoolMeter const &getMeter() const;
-
-    /**
-     * \param stats Object to be filled with statistical data about pool.
-     * \retval      Number of objects in use, ie. allocated.
-     */
-    int getStats(MemPoolStats * stats);
-
-    void zeroBlocks(bool doIt);
+    /* Mem::Allocator API */
```

Comment:
If this class survives, please consider adding either `final override` or `override` to these "new" API overrides.

Reply:
Will do after the prerequisite PRs are merged.